modules/improvedigitalBidAdapter.js

```diff
@@ -192,539 +265,454 @@ export const spec = {
-  bid.renderer.handleVideoEvent({ id, eventName });
-}
+const ID_UTILITY = {
+  getLanguage() {
+    return navigator.language.split('-')[0]
+  },
 
-function createRenderer(bidRequest) {
-  const renderer = Renderer.install({
-    id: bidRequest.adUnitCode,
-    url: RENDERER_URL,
-    loaded: false,
-    config: deepAccess(bidRequest, 'renderer.options'),
-    adUnitCode: bidRequest.adUnitCode
-  });
-  try {
-    renderer.setRender(outstreamRender);
-  } catch (err) {
-    logWarn('Prebid Error calling setRender on renderer', err);
-  }
-  return renderer;
-}
+  isInstreamVideo(bid) {
+    const context = deepAccess(bid, 'mediaTypes.video.context');
+    return bid.mediaType === 'video' || context !== 'outstream';
+  },
 
-function getNormalizedBidRequest(bid) {
-  let adUnitId = getBidIdParameter('adUnitCode', bid) || null;
-  let placementId = getBidIdParameter('placementId', bid.params) || null;
-  let publisherId = null;
-  let placementKey = null;
+  isOutstreamVideo(bid) {
+    return deepAccess(bid, 'mediaTypes.video.context') === 'outstream';
+  },
 
-  if (placementId === null) {
-    publisherId = getBidIdParameter('publisherId', bid.params) || null;
-    placementKey = getBidIdParameter('placementKey', bid.params) || null;
-  }
-  const keyValues = getBidIdParameter('keyValues', bid.params) || null;
-  const singleSizeFilter = getBidIdParameter('size', bid.params) || null;
-  const bidId = getBidIdParameter('bidId', bid);
-  const transactionId = getBidIdParameter('transactionId', bid);
-  const currency = config.getConfig('currency.adServerCurrency');
-
-  let normalizedBidRequest = {};
-  if (isInstreamVideo(bid)) {
-    normalizedBidRequest.adTypes = [ VIDEO ];
-  }
-  if (isInstreamVideo(bid) || isOutstreamVideo(bid)) {
-    normalizedBidRequest.video = getVideoTargetingParams(bid);
-  }
-  if (placementId) {
-    normalizedBidRequest.placementId = placementId;
-  } else {
-    if (publisherId) {
-      normalizedBidRequest.publisherId = publisherId;
+  getBidFloor(bid) {
+    if (!isFn(bid.getFloor)) {
+      return null;
     }
-    if (placementKey) {
-      normalizedBidRequest.placementKey = placementKey;
+    const floor = bid.getFloor({
+      currency: 'USD',
+      mediaType: '*',
+      size: '*'
+    });
+    if (isPlainObject(floor) && !isNaN(floor.floor) && floor.currency === 'USD') {
+      return floor.floor;
     }
-  }
-
-  if (keyValues) {
-    normalizedBidRequest.keyValues = keyValues;
-  }
+    return null;
+  },
 
-  if (config.getConfig('improvedigital.usePrebidSizes') === true && !isInstreamVideo(bid) && !isOutstreamVideo(bid) && bid.sizes && bid.sizes.length > 0) {
-    normalizedBidRequest.format = bid.sizes;
-  } else if (singleSizeFilter && singleSizeFilter.w && singleSizeFilter.h) {
-    normalizedBidRequest.size = {};
-    normalizedBidRequest.size.h = singleSizeFilter.h;
-    normalizedBidRequest.size.w = singleSizeFilter.w;
-  }
+  outstreamRender(bid) {
```

Comment:
please move all methods related to outstreaming rendering from ID_UTILITY into a new object ID_OUTSTREAM. Then "outstream" can be removed from method name.